sql-jdbc/src/main/java/org/opensearch/jdbc/ResultSetImpl.java

```diff
@@ -1071,14 +1075,32 @@ public Date getDate(String columnLabel, Calendar cal) throws SQLException {
 
     @Override
     public Time getTime(int columnIndex, Calendar cal) throws SQLException {
-        // TODO - implement?
-        return null;
+        log.debug(() -> logEntry("getTime (%d, %s)", columnIndex,
+                cal == null ? "null" : "Calendar TZ= " + cal.getTimeZone()));
+        checkCursorOperationPossible();
+        log.debug(() -> {
+            try {
+                return logExit("getTime", getTimeX(columnIndex, cal));
```

Comment:
Whoops.  This is not right.  I misunderstood, and thought getTimeX was only needed for debug calls... but we certainly don't want this.  We should only call getTimeX once.  Please revert this back to what you had before.